src/main/scala/geotrellis/sdg/PopulationNearRoadsLookupJob.scala

```diff
@@ -0,0 +1,264 @@
+      val qaTiles = OsmQaTiles.fetchFor(country)
+      val row = qaTiles.fetchRow(key).get
+      val roads: Seq[MultiLineString] = PopulationNearRoadsLookupJob.extractRoads(row.tile, roadFilter)
+      val buffered = roads.map(GeometryUtils.bufferByMeters(_, WebMercator, layoutTileSource.source.crs, meters = 2000))
+      PopulationNearRoadsLookupJob.burnRoadMask(layoutTileSource.layout, buffered)
+    }
+  }.reduceByKey(partitioner, (l, r) => PopulationNearRoadsLookupJob.combineMasks(l, r))
+
+  // We have RDD of countries and COG keys we will need to read.
+  // However, there could be pop regions NOT covered by COGs that we still need to read
+  // So we should consider all regions available form WorldPop and join them to vectors
+
+  // TODO: try read and tile approach for performance
+  val grumpMaskRdd: RDD[(SpatialKey, Tile)] =
+    Grump.masksForBoundary(grumpRdd, layout, wsCountryBorder, partitioner)
+      .setName(s"${country.code} GRUMP Mask")
+
+  val popRegions: RDD[(SpatialKey, SummaryRegion)] =
+    regionsRdd.
+      cogroup(roadMaskRdd, grumpMaskRdd, partitioner).
+      flatMap { case (key, (_, roadMasks, grumpMasks)) =>
+        // !!! This part is CRITICAL !!!
+        // We moved the key and not the RasterRegion because each regions will spawn a new RasterSource
+        // This would result in new S3Client and Metadata fetch for EVERY segment read.
+        // So we generate RasterRegion from key inside the map step
+        layoutTileSource.rasterRegionForKey(key).
+          map({ region =>
+            val sm = SummaryRegion(region, roadMasks.headOption, grumpMasks.headOption)
+            (key, sm)
+          })
+      }
+
+  val forgottenLayer: TileLayerRDD[SpatialKey] = {
+    val rdd = popRegions.flatMap { case (key, region) =>
+      region.forgottenPopTile.map(tile => (key, tile))
+    }
+
+    // This is the first time in the job flow we're trying to read COG on driver, log for tracing
+    logger.info(s"Reading COG: ${rasterSource.name}")
+    val md = TileLayerMetadata(rasterSource.cellType, layout, rasterSource.extent,
+      rasterSource.crs, KeyBounds(layout.mapTransform.extentToBounds(rasterSource.extent)))
+
+    ContextRDD(rdd, md)
+  }
+
+  def forgottenLayerTiles(outputUri: URI, pixelScale: Int): Unit = {
+    import spark.implicits._
+    val maxZoom = 12
+    val minZoom = 6
+    // Tweak this value by powers of 2 to increase or reduce the pixel size in the output
+    // Higher number == smaller pixels
+    val pixelsPerTile = math.pow(2, pixelScale).toInt
+    val wmLayoutScheme = ZoomedLayoutScheme(WebMercator, pixelsPerTile)
+    val wmLayout: LayoutDefinition = wmLayoutScheme.levelForZoom(maxZoom).layout
+    val latLngToWebMercator = Transform(LatLng, WebMercator)
+
+    val gridPointsRdd: RDD[(Long, Long, Double)] = forgottenLayer.flatMap {
+      case (key: SpatialKey, tile: Tile) => {
+        val h3: H3Core = H3Core.newInstance
```

Comment:
There's a bad merge in here. And it's pretty wigged out. The new code uses the H3Core library to calculate indices and pass along lon, lat, index. The yield below still uses the old code. Probably want to just copy/paste this whole `def forgottenLayerTiles` method from `master` branch.

Reply:
Yeah, looked into this a bit further - this code path was no longer being exercised because this whole vectortile production process is a static method. It looks as though I simply failed to remove this method from one of the two implementing `Job` classes.